cmd/cloud-controller-manager/app/controllermanager.go

```diff
@@ -299,6 +248,32 @@ func startControllers(c *cloudcontrollerconfig.CompletedConfig, stop <-chan stru
 	select {}
 }
 
+// IsControllerEnabled check if a specified controller enabled or not.
+func IsControllerEnabled(name string, disabledByDefaultControllers sets.String, controllers ...string) bool {
```

Comment:
isn't the flag implementation generic? Why is this logic here and not where the flag is defined?

Reply:
Follow KCM. may be we can move it to `cmd/controller-manger` , so we can reuse it both in KCM and CCM

Reply:
That would be ideal but a little more refactoring will be required. Might be cleaner to have that in another PR, just my 2 cents :)

Reply:
fixed, move `IsControllerEnabled()` into `cmd/controller-manager/app/helper.go`